tests/cpp/test_alias.cpp

```diff
@@ -1637,4 +1638,35 @@ TEST_F(AliasTest, IntermediateTensorWithAllocation) {
           HeuristicIs(SchedulerType::ExprEval)));
 }
 
+TEST_F(AliasTest, SegmentSetRetainsSharding) {
+  Fusion fusion;
+  FusionGuard fg(&fusion);
+
+  DeviceMesh mesh({0, 1});
+  TensorView* tv0 = makeContigTensor(2);
+  TensorView* tv1 = add(tv0, tv0);
+  TensorView* tv2 = set(tv1);
+  fusion.addInput(tv0);
+  fusion.addOutput(tv1);
+  fusion.addOutput(tv2);
+
+  for (auto* tv : {tv0, tv1, tv2}) {
+    tv->setDeviceMesh(mesh);
+    tv->outer_split(0, mesh.size());
+    tv->axis(0)->parallelize(ParallelType::DIDx);
+  }
+
+  preseg_passes::OptimizationPass<
+      preseg_passes::MarkAliasesPreparePass>::runPass(&fusion);
```

Comment:
https://testing.googleblog.com/2013/08/testing-on-toilet-test-behavior-not.html

To make the test more realistic, strict and robust, I'd do the following
1. Run the fusion on multiple GPUs and therefore move this test to bin/test_multidevice
2. Verify the numerical result
3. Check none of the segments are SchedulerType::Communication. You can `Each(Not(HeuristicsIs(SchedulerType::Communication)))` for that
4. Optionally, check runtime->fusionSegments()->completeFusion() has a SegmenterSet. I'm fine with not checking that because SegmenterSet is an implementation detail.

Reply:
If you do this, remove the mark_aliases_prepare_pass.h include.